hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
-    Collection<Condition> getConditions(String triggerId) throws Exception;
+
+    /**
+     * A convenience method that adds a new Condition to the existing condition set for the specified
+     * Trigger and trigger mode.  The new condition will be assigned the highest conditionSetIndex for the
+     * updated conditionSet. The following Condition fields are ignored for the
+     * incoming condition, and set in the returned collection set:<pre>
+     *   conditionId
+     *   triggerId
+     *   triggerMode
+     *   conditionSetSize
+     *   conditionSetIndex
+     * </pre>
+     * @param triggerId
+     * @param triggerMode
+     * @param condition Not null
+     * @return The updated, persisted condition set
+     * @throws Exception
+     */
+    Collection<Condition> addCondition(String triggerId, Trigger.Mode triggerMode, Condition condition)
+            throws Exception;
+
+    /**
+     * A convenience method that removes a Condition from an existing condition set. This will update the
+     * conditionSetSize and possibly the conditionSetIndex for any remaining conditions.
+     * @param conditionId
+     * @return The updated, persisted condition set. Not null. Can be empty.
+     * @throws Exception
+     */
+    Collection<Condition> removeCondition(String conditionId)
+            throws Exception;
+
+    /**
+     * A convenience method that updates an existing Condition from an existing condition set.
+     * @param condition Not null. conditionId must be for an existing condition.
+     * @return The updated, persisted condition set. Not null. Can be empty.
+     * @throws Exception
+     */
+    Collection<Condition> updateCondition(Condition condition)
+            throws Exception;
+
+    /**
+     * The condition set for a trigger's trigger mode is treated as a whole.  When making any change to the
+     * conditions just [re-]set all of the conditions. The following Condition fields are ignored for the
+     * incoming conditions, and set in the returned collection:<pre>
+     *   conditionId
+     *   triggerId
+     *   triggerMode
+     *   conditionSetSize
+     *   conditionSetIndex
+     * </pre>
+     * @param triggerId
+     * @param triggerMode
+     * @param conditions Not null, Not Empty
+     * @return The persisted condition set
+     * @throws Exception
+     */
+    Collection<Condition> setConditions(String triggerId, Trigger.Mode triggerMode, Collection<Condition> conditions)
+            throws Exception;
+
     Condition getCondition(String conditionId) throws Exception;
 
+    /**
+     * @param triggerId
+     * @param triggerMode Return only conditions for the given trigger mode. Return all if null.
+     * @return The existing conditions for the trigger. Not null.
+     * @throws Exception
+     */
+    Collection<Condition> getTriggerConditions(String triggerId, Trigger.Mode triggerMode) throws Exception;
+
+    Collection<Condition> getAllConditions() throws Exception;
```

Comment:
Also, same thought for Triggers in relation to tagging/groups is valid here with conditions.
By default, a Trigger is not attached to a resource (what is good IMO) as we can have flexibility to define triggers with multiple resources/metrics. Also I'm thinking if the tagging/group mechanism can also be applied to conditions, so, we can get all conditions of a specific tag, with no matters in which triggers depends on.
So, in theory this is interesting for operations when we may want to apply a template/conditions to a groups of resources (at least without enter in low level details it can be a start point to refine later).